core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -313,6 +314,35 @@ final class ShuffleBlockFetcherIterator(
   }
 }
 
+// Helper class that ensures a ManagerBuffer is released upon InputStream.close()
+private class WrappedInputStream(delegate: InputStream, iterator: ShuffleBlockFetcherIterator)
```

Comment:
What about renaming this to BufferReleasingInputStream or similar? I'm concerned that there are a bunch of other classes in Spark that are a wrapped input stream, so specificity in the class name is helpful.